src/sentry/api/authentication.py

```diff
@@ -167,3 +168,19 @@ def authenticate_credentials(self, token):
             scope.set_tag("api_project_key", key.id)
 
         return (AnonymousUser(), key)
+
+
+class SystemKeyAuthentication(StandardAuthentication):
+    token_name = b'system'
+
+    def authenticate_credentials(self, token):
+        correct_token = options.get('system.secret-key')
+        assert correct_token, "Sentry should have never been able to start without a secret key configured"
+
+        if token != correct_token:
```

Comment:
Should this be a constant time comparison?

Reply:
Good catch, yes it should!